### PR TITLE
Support for Auto-detecting FST type, and adding type hint to FST files when packaged

### PR DESCRIPTION
**Pull request by ZappoMan**
* support for auto-detecting FST type when installing an avatar from the Marketplace.
* support for including the type-hint in the FST when packaging the content
* classifying FSTReader

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3bda65f4fe9a9cd3e17daa53fa686f8ea8f0f100 available at: https://gist.github.com/9dac0386ed8886aefd84<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4528.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4528.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -3645,17 +3645,56 @@ bool Application::askToSetAvatarUrl(const QString& url) {
         msgBox.exec();
         return false;
     }
-
-    QString message = "Would you like to use this model for part of avatar:\n" + url;
+    
+    // Download the FST file, to attempt to determine it's model type
```
Comment (sethalves) [inline interface/src/Application.cpp]: wild apostrophe

File: interface/src/ModelPackager.cpp [hunk 2]
```diff
@@ -135,6 +136,7 @@ bool ModelPackager::editProperties() {
         
         return false;
     }
+    */
```
Comment (sethalves) [inline interface/src/ModelPackager.cpp]: why leave this commented out?

File: libraries/fbx/src/FSTReader.cpp [hunk 3]
```diff
@@ -10,10 +10,11 @@
 //
 
 #include <QBuffer>
+#include <QDebug>
```
Comment (sethalves) [inline libraries/fbx/src/FSTReader.cpp]: not longer need QDebug?

File: interface/src/ModelPackager.cpp (hunk 2 above)
Comment (ZappoMan) [inline interface/src/ModelPackager.cpp]: gak! Didn't mean to leave this in... will look at it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0366ce9bf5f885d2a27813c70e2362e328213023 available at: https://gist.github.com/9dac0386ed8886aefd84<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4528.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4528.dmg

**Pull request merged**